x/ccv/provider/keeper/throttle.go

```diff
@@ -0,0 +1,420 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/x/ccv/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// This file contains functionality relevant to the throttling of slash and vsc matured packets, aka circuit breaker logic.
+
+// HandlePendingSlashPackets handles all or some portion of pending slash packets received by consumer chains,
+// depending on throttling logic. The slash meter is decremented appropriately in this method.
+func (k Keeper) HandlePendingSlashPackets(ctx sdktypes.Context) {
```

Comment:
The method name is misleading because it handles pending slash packets, but it also handles pending maturity packets which are behind the slashes.

Reply:
Agreed, will change